src/components/Box/Box.tsx

```diff
@@ -1,30 +1,62 @@
 import * as React from 'react'
+import classnames from 'classnames'
+import arrowDownIcon from '../../assets/arrow-down.svg'
+import arrowUpIcon from '../../assets/arrow-up.svg'
+import { Button } from '../Button/Button'
 import './Box.css'
 
 export type BoxProps = {
   header?: string
   className?: string
   children: React.ReactNode
   borderless?: boolean
+  collapsible?: boolean
+  collapsed?: boolean
+  onToggle?: (shouldCollapse: boolean) => void
 }
 
-export class Box extends React.PureComponent<BoxProps> {
-  render(): JSX.Element {
-    const { className, header, children, borderless } = this.props
+export const Box = (props: BoxProps): JSX.Element => {
+  const {
+    collapsed,
+    header,
+    children,
+    borderless,
+    className,
+    collapsible,
+    onToggle
+  } = props
+  const [isCollapsed, setIsCollapsed] = React.useState(
+    'collapsed' in props ? collapsed : false
+  )
 
-    const classes = ['dcl', 'box']
-    if (className) {
-      classes.push(className)
-    }
-    if (borderless) {
-      classes.push('borderless')
-    }
+  const shouldCollapse = 'collapsed' in props ? collapsed : isCollapsed
```

Comment:
Is this `shouldCollapse` variable needed? `isCollapsed` has the  `'collapsed' in props` check so maybe we can use that one below